### PR TITLE
Issue91 more tests

### LINKED ISSUE #91: Increase test coverage
At the moment, the tutorial is also used for testing. This should be avoided, and the proper tests should be added to the `nansat.tests` package.

### PR DESCRIPTION
**Pull request by akorosov**
@korvinos , can you have a look?
I suggest to close #91 after this PR is merged and open more specific issues.

### CODE REVIEW

File: nansat/tests/test_vrt.py [hunk 1]
```diff
@@ -545,5 +557,44 @@ def test_get_super_vrt_and_copy(self):
...
+        with warnings.catch_warnings(record=True) as w:
+            vrt_filename = vrt.fileName
+            self.assertEqual(vrt_filename, vrt.filename)
+
+    def test_get_sub_vrt0(self):
+        vrt1 = VRT()
+        vrt2 = vrt1.get_sub_vrt()
+        self.assertEqual(vrt1, vrt2)
+
+    def test_get_sub_vrt3(self):
```
Comment (korvinos) [inline nansat/tests/test_vrt.py]: When I tried to understand what is going on I went to docstring of get_super_vrt method and found: `copy of self in vrt.vrt and change references from vrt to vrt.vrt`. Seems like this is very important method, so could you please include a better documentation of the <get_super_vrt> method to this PR

File: nansat/tests/test_vrt.py [hunk 2]
```diff
@@ -545,5 +557,44 @@ def test_get_super_vrt_and_copy(self):
         self.assertFalse(data is None)
         self.assertTrue(np.all(data == array))
 
+    def test_filename_warning(self):
+        vrt = VRT()
+        with warnings.catch_warnings(record=True) as w:
```
Comment (korvinos) [inline nansat/tests/test_vrt.py]: You do not test the warning here

File: nansat/vrt.py [hunk 3]
```diff
@@ -1496,12 +1494,7 @@ def transform_points(self, col_vector, row_vector, dst2src=0,
         lonlat = transformer.TransformPoints(dst2src, xy)[0]
```
Comment (korvinos) [inline nansat/vrt.py]: what is lonlat? Could you add a comment, please? (Description of structure or something)

File: nansat/vrt.py [hunk 4]
```diff
@@ -1496,12 +1494,7 @@ def transform_points(self, col_vector, row_vector, dst2src=0,
...
 
         # convert return to lon,lat vectors
-        lonlat = np.array(lonlat)
-        if lonlat.shape[0] > 0:
-            lon_vector = lonlat[:, 0]
-            lat_vector = lonlat[:, 1]
-        else:
-            lon_vector, lat_vector = [], []
```
Comment (korvinos) [inline nansat/vrt.py]: Won't the method be able to return an empty lons and lats? If you could add more information to the previous question (line 1496) it would be more clear

File: nansat/vrt.py [hunk 5]
```diff
@@ -1064,8 +1064,6 @@ def create_band(self, src, dst=None):
             srcs = [src]
         elif type(src) in [list, tuple]:
             srcs = src
-        else:
-            raise ValueError('Wrong src type (%s)! Should be dict or list/tuple of dict'%type(src))
```
Comment (korvinos) [inline nansat/vrt.py]: Why did you remove the Exception? It is potentially possible to get a wrong data type

File: nansat/tests/test_vrt.py (hunk 1 above)
Comment (akorosov) [reply nansat/tests/test_vrt.py]: Fixed in 96fbd90

File: nansat/tests/test_vrt.py (hunk 2 above)
Comment (akorosov) [reply nansat/tests/test_vrt.py]: Fixed in 96fbd90

File: nansat/vrt.py (hunk 3 above)
Comment (akorosov) [reply nansat/vrt.py]: Fixed in 96fbd90

File: nansat/vrt.py (hunk 4 above)
Comment (akorosov) [reply nansat/vrt.py]: Fixed in 96fbd90

File: nansat/vrt.py (hunk 5 above)
Comment (akorosov) [reply nansat/vrt.py]: It is potentially possible to give wrong input in any method. To keep it simple I suggest not to add checks of inputs without strict necessity.
